zerver/lib/upload.py

```diff
@@ -147,6 +151,18 @@ def get_realm_for_filename(path):
         return None
     return get_user_profile_by_id(key.metadata["user_profile_id"]).realm.id
 
+def delete_message_image_s3(path_id):
+    conn = S3Connection(settings.S3_KEY, settings.S3_SECRET_KEY)
+    bucket = get_bucket(conn, settings.S3_AUTH_UPLOADS_BUCKET)
+
+    #check if file exists
```

Comment:
I think this code path should log a warning if the file doesn't exist, since we should only be calling this function on files that do exist.